tests/apis/parameters.spec.ts

```diff
@@ -0,0 +1,139 @@
+import { expect } from "chai";
+import { extend } from "lodash";
+import { createSandbox, stub } from "sinon";
+import { getParameter } from '../../src/apis/parameters';
+
+describe("getParameter", () => {
+    let sandbox: any;
+    const WS_BASE_URL = "WS-BASE-VALUE";
+    const parameterName = "param1";
+
+    before(() => {
+        sandbox = createSandbox();
+    });
+    
+    after(() => {
+        sandbox.restore();
+    });
+
+    // it("not on BAS ---> returns undefined", async () => {
+    //     stubEnv({ WS_BASE_URL: "" });
+    //     const parameterValue = await getParameter(parameterName);
+    //     expect(parameterValue).to.be.undefined;
+    // });
+
+    it("no @sap/plugin loaded ---> returns undefined", async () => {
+        stubEnv({ WS_BASE_URL: WS_BASE_URL });
+        const parameterValue = await getParameter(parameterName);
+        expect(parameterValue).to.be.undefined;
+    });
+
+    it("@sap/plugin loaded, configuration is null ---> returns undefined", async () => {
+        stubEnv({ WS_BASE_URL: WS_BASE_URL });
+        const requireMock = require('mock-require');
+
+        const sapPlugin = {
+            window: {
+                configuration: () => null
+            }
+        };            
+        requireMock('@sap/plugin', sapPlugin);
+
+        const parameterValue = await getParameter(parameterName);
+        expect(parameterValue).to.be.undefined;
+        requireMock.stop('@sap/plugin');
+    });
+
+    it("@sap/plugin loaded, configuration is undefined ---> returns undefined", async () => {
+        stubEnv({ WS_BASE_URL: WS_BASE_URL });
+        const requireMock = require('mock-require');
+
+        const sapPlugin = {
+            window: {
+                configuration: () => undefined

```

Comment:
inspect what is possible according to `window.configuration()` API.
and consider which edge cases need to be checked.

Reply:
The API:
  export namespace window {
    export function configuration(): Promise<{ [id: string]: string }>;
  }
They has initialization to empty in the code:
const configuration: { [id: string]: string } = {};
So currently it can be only empty and not null and not undefined.
? behaves the same on null and undefined
So I am keeping only null configuration.